email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -1,6 +1,82 @@
+     * send the email and get its message ID, obtain a completable future as in the following example.
+     * <pre>{@code
+     * String messageID = completionStage.toCompletableFuture().get();
+     * }</pre>
+     * Checked exceptions that occur while sending the email are instances of {@link EmailSenderException} that are
+     * wrapped in an {@link ExecutionException} thrown by {@link CompletableFuture#get()}.
+     * <p>To send e-mails synchronously, use {@link #sendAndWait(MessageAuthor, int, TimeUnit)}.
+     *
+     * @param messageAuthor a message author that fills the empty message created by the email sender.
+     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the message
+     *                      author must only be used by that single author instance and must not be passed elsewhere.
+     * @return A completion stage for sending the message asynchronously.
+     * @see #sendAndWait(MessageAuthor, int, TimeUnit)
+     */
+    @Nonnull
+    CompletionStage<String> send(@Nonnull final MessageAuthor messageAuthor);
+
+    /**
+     * Send the given message and wait for the result.
+     * <p>
+     * This method is provided as a convenience method to users who send e-mails synchronously. It is invoked in the
+     * same way as {@link #send(MessageAuthor)}, but with an additional timeout. Note that the choice of the
+     * timeout influences how your application performs under heavy load and in the case of failure of the e-mail
+     * infrastructure.
+     *
+     * @param messageAuthor a message author that fills the empty message created by the email sender.
+     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the message
+     *                      author must only be used by that single author instance and must not be passed elsewhere.
+     * @param timeout       the amount of time to wait until a {@link TimeoutException} is raised
+     * @param unit          the unit of the time amount
+     * @return the ID of the successfully sent message (this method will not return otherwise)
+     * @throws ExecutionException   if there was an exception while sending the e-mail. The wrapped exception will be a
+     *                              {@link EmailSenderException} if the e-mail could not be sent due to issues with the
+     *                              given message or the e-mail infrastructure.
+     * @throws TimeoutException     if the timeout elapsed
+     * @throws InterruptedException if the current thread was interrupted while waiting for the e-mail to be sent
+     * @see #send(MessageAuthor)
+     */
+    default
+    @Nonnull
+    String sendAndWait(@Nonnull final MessageAuthor messageAuthor, final int timeout, @Nonnull final TimeUnit unit)
+            throws ExecutionException, TimeoutException, InterruptedException {
```

Comment:
here I would also prefer to only receive runtime exceptions.
It would be okay for now to catch those exceptions and may wrap them or the causse of ExecutionException in a CompletionException.